simplekv/cache.py

```diff
@@ -110,6 +110,32 @@ def open(self, key):
             # cache error, ignore completely and return from backend
             return self._dstore.open(key)
 
+    def _copy(self, source, dest):
+        """Implementation of :meth:`~simplekv.CopyMoveMixin.copy`.
+        
+        Copies the data in the backing store and removes the destination key from the cache,
+         in case it was already populated.
+         """
+        try:
+            k = self._dstore._copy(source, dest)
+        finally:
+            self.cache.delete(dest)
+        return k
+
+    def _move(self, source, dest):
+        """Implementation of :meth:`~simplekv.CopyMoveMixin.move`.
+        
+        Moves the data in the backing store and removes both the source and destination key from the cache.
+        """
+        try:
+            k = self._dstore._move(source, dest)
+        finally:
+            try:
+                self.cache.delete(source)
+            finally:
+                self.cache.delete(dest)
+        return k
+
```

Comment:
With a cache, there are potentially three things that might or might not have the copy/move capability: the base store, the cache store, and the combined object. 

I'll use this comment to put some of my thoughts into writing, hoping to get them in order eventually ;-)

- We could have four types of `CacheDecorator`s, one for each possible combination. This feels wrong to me.
- We could have just one `CacheDecorator`, which has `.copy()`/`.move()` methods which try using the functionality in the cache- and base stores, if available, and fall back to the dumb implementation if not. This seems wrong to me as well, because it doesn't make sense that a `CacheDecorator` would have methods that are not in the core API.
- Another way would be to add a `CopyMoveCacheDecorator`, which has `.copy()`/`.move()` and uses `.copy()` and `.move()` of the cache- and base store respectively, if available, and otherwise falls back to the 'dumb' implementation. This also feels messy to me.

Honestly, unless we can think of a cleaner solution, my preference would be to leave copy/move out of the cached stores. The current solution just implicitly expects the functionality to be there and fails at runtime when it is not. If I am not overlooking something, a store that benefits from a cache tends to be mostly or exclusively read, while a store that needs copy/write functionality is needed for more "maintenance-y" tasks.